src/hmm/hmmer.py

```diff
@@ -0,0 +1,477 @@
+
+                # if there is no data, input_data will be set to None
+                # this tells the worker to shut down
+                # if there is, it's a list that can be sent to the worker directly
+                connection.send(input_data)
+
+                # we do need to clean up the connection if it is going to close, though
+                if input_data is None:
+                    connection.close()
+                    connections.remove(connection)
+
+                # now lets handle the output data
+                # it will be either None or a list
+                # the fist element of the list will be a number indicating the amount of
+                # tasks this output came from
+                # any other elements are a tuple of format:
+                # (cds_id, score, domain_accession)
+                # we only care about if a list is returned. otherwise we just move on
+                if output_data is not None:
+                    src_task_count: int = output_data[0]
+                    task_outputs: list[tuple[int, str, float]] = output_data[1:]
+                    for task_output in task_outputs:
+                        yield task_output_to_hsp(task_output, genes)
+
+                    tasks_done += src_task_count
+                    if callback is not None:
+                        callback(tasks_done)
+
+        # just to make sure, kill any remaining processes
+        for process in processes:
+            process.kill()
+
+    @staticmethod
+    def profile_hmmsearch(
+        sequences: list[TextSequence],
+    ) -> list[tuple[int, str, float, int, int]]:
+        """Performs search_hmm on the given list of sequences for each profile in
+        HMMer.profiles
+
+        Args:
+            sequences (list[TextSequence]): list of input sequences
+
+        Returns:
+            list[tiple[int, str, float]]: result object to pass back to the main thread
+        """
+        text_seq_block = TextSequenceBlock(sequences).digitize(HMMer.alphabet)
+
+        outputs = []
+        # perform hmmsearch for each profile
+        for profile in HMMer.profiles:
+            hmmscan_results = HMMer.pipeline.search_hmm(profile, text_seq_block)
+            for top_hit in hmmscan_results.reported:
+                for domain in top_hit.domains:
+                    if not domain.reported:
+                        continue
+
+                    # name is actually the list index of the original CDS
+                    cds_idx = int(top_hit.name.decode())
+                    accession = domain.alignment.hmm_accession.decode()
+                    score = domain.score
+                    env_start = domain.env_from
+                    env_stop = domain.env_to
+                    outputs.append((cds_idx, accession, score, env_start, env_stop))
+
+        return outputs
+
+    @staticmethod
+    def hmmsearch_process(process_id, connection: Connection):  # pragma: no cover
```

Comment:
forgot to talk about this. coverage reports do not see this function being covered because it is being run in a different process. pragma: no cover is telling coverage to not care about this function when it calculates coverage